libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -414,6 +414,11 @@ bool AC_Fence::pre_arm_check_polygon(char *failure_msg, const uint8_t failure_ms
 // additional checks for the circle fence:
 bool AC_Fence::pre_arm_check_circle(char *failure_msg, const uint8_t failure_msg_len) const
 {
+    if (!(_configured_fences & AC_FENCE_TYPE_CIRCLE)) {
```

Comment:
The check is a sanity check _prior_ to vehicle arming. If one was to enable or make use of circular fences during flight, and `_circle_radius` is not correctly set, what is the resultant behaviour?

I would tend to prefer to have sane values set whether or not I am actively making use of the fence.

Reply:
I believe that performing health checks on unused parameters is unnecessary.
Even if an anomaly is detected in the health check of an unused parameter, it has no impact on operation.
I would like anomaly notifications to be limited to issues that actually affect operation.

Reply:
> The check is a sanity check _prior_ to vehicle arming. If one was to enable or make use of circular fences during flight, and `_circle_radius` is not correctly set, what is the resultant behaviour?

There are *many* parameters in ArduPilot that you could change at runtime and get anomalous behaviour.  Perhaps if the default values of variables would be problematic we should care, but I'm not sure this is such a case.

Is there a workflow you're thinking of where a user would go and enable these parameters in-flight go get some sort of behaviour - and where they wouldn't already have considered the radius ahead of time?

Not unrelated to the fact that you can say, "honour polygon fences" and not actually have any, BTW.

Reply:
When the polygon fence is enabled, the presence of the polygon fence is being checked.
There are cases where only the polygon fence is enabled.
Other fences should be treated the same way as the polygon fence.

    if (!(_configured_fences & AC_FENCE_TYPE_POLYGON)) {
        // not enabled; all good
        return true;
    }

Reply:
Just to be clear, I'm certainly not holding this up. And struggle to think of a case where it might make less sense.

I think the catch is, you can have configured fences, but not have them enabled. In that case, you should be advised when your configured fence has an issue.